### PR TITLE
perf triage for 2021-07-20.

### PR DESCRIPTION
**Pull request by pnkfelix**

### CODE REVIEW

File: triage/2021-07-20.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+3 Regressions, 3 Improvements, 3 Mixed; 1 of them in rollups
+
+#### Regressions
+
+Replace associated item bound vars with placeholders when projecting [#86993](https://github.com/rust-lang/rust/issues/86993)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=b1f8e27b74c541d3d555149c8efa4bfe9385cd56&end=27e42058811e448b1a7dd8630d86ab247fbfcb9b&stat=instructions:u) (up to 1.5% on `incr-patched: b9b3e592dd cherry picked` builds of `style-servo-debug`)
+- A perf run was [done on the PR originally](https://perf.rust-lang.org/compare.html?start=fdfe819580062a441024d713b49340cd3f7d7efc&end=7baa78ec683fd14db4d4c1869dbef5cbbc5b774d). The regression flagged here seems to be on a different set of benchmarks: `style-servo-debug` and `wf-projection-stress-65510-*`, neither of which was flagged as a regression in the original runs.
+- Results overall seem pretty mixed and hard to interpret. Left a [comment](https://github.com/rust-lang/rust/pull/86993#issuecomment-883624762) on the PR.
```
Comment (rylev) [inline triage/2021-07-20.md:15]: I'm not sure why the triage report labeled this as a regression when the results seem so mixed... I will investigate this.

File: triage/2021-07-20.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Remove special case for `ExprKind::Paren` in `MutVisitor` [#87284](https://github.com/rust-lang/rust/issues/87284)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=014026d1a7ca991f82f12efa95ef4dffb29dc8af&end=6535449a002264ee08dec8e741f1aadd97428fae&stat=instructions:u) (up to 1.0% on `full` builds of `coercions-debug`)
+- This is on the borderline for "significant regression", and was the *only* regression associated with this PR.
```
Comment (rylev) [inline triage/2021-07-20.md:19]: Eventually I'd like to make the logic for labeling runs as regressions to be a bit smarter in these cases. We should be able to encode that if there is only a single noisy benchmark which regressed and that regression is barely significant, we shouldn't count it as a regression.

File: triage/2021-07-20.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Triage done by **@pnkfelix**.
+Revision range: [5aff6dd07a562a2cba3c57fc3460a72acb6bef46..5c0ca08c662399c1c864310d1a20867d3ab68027](https://perf.rust-lang.org/?start=5aff6dd07a562a2cba3c57fc3460a72acb6bef46&end=5c0ca08c662399c1c864310d1a20867d3ab68027&absolute=false&stat=instructions%3Au)
+
+3 Regressions, 3 Improvements, 3 Mixed; 1 of them in rollups
+
+#### Regressions
+
+Replace associated item bound vars with placeholders when projecting [#86993](https://github.com/rust-lang/rust/issues/86993)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=b1f8e27b74c541d3d555149c8efa4bfe9385cd56&end=27e42058811e448b1a7dd8630d86ab247fbfcb9b&stat=instructions:u) (up to 1.5% on `incr-patched: b9b3e592dd cherry picked` builds of `style-servo-debug`)
+- A perf run was [done on the PR originally](https://perf.rust-lang.org/compare.html?start=fdfe819580062a441024d713b49340cd3f7d7efc&end=7baa78ec683fd14db4d4c1869dbef5cbbc5b774d). The regression flagged here seems to be on a different set of benchmarks: `style-servo-debug` and `wf-projection-stress-65510-*`, neither of which was flagged as a regression in the original runs.
```
Comment (rylev) [inline triage/2021-07-20.md:14]: It seems there was no comment on the original runs in the PR - we should encourage people to comment on the runs for those look at the issue in the future.

File: triage/2021-07-20.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+- It seems like these results were [expected](https://github.com/rust-lang/rust/pull/87044#issuecomment-879407381), and they make sense given the nature of the PR.
+
+Update Rust Float-Parsing Algorithms to use the Eisel-Lemire algorithm. [#86761](https://github.com/rust-lang/rust/issues/86761)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=64d171b8a419eb6cb872ab579398eff8a741bbc6&end=f502bd3abd12111bbfae0974db018c165a977c0e&stat=instructions:u) (up to 3.2% on `incr-patched: b9b3e592dd cherry picked` builds of `style-servo-debug`)
+- Moderate improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=64d171b8a419eb6cb872ab579398eff8a741bbc6&end=f502bd3abd12111bbfae0974db018c165a977c0e&stat=instructions:u) (up to -1.4% on `full` builds of `piston-image-opt`)
+- A perf run was done on the orignal PR, but the run there did not predict the regression that was now observed to `style-servo-debug` when this PR landed on nightly.
+- Left a comment, but I am guessing that `style-servo-debug` may need a bigger noise threshold (or rather, sensitivity to perturbations) than what we currently use.
```
Comment (rylev) [inline triage/2021-07-20.md:53]: nit: link https://github.com/rust-lang/rust/pull/86761#issuecomment-883645181

File: triage/2021-07-20.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+Move OnDiskCache to rustc_query_impl. [#86698](https://github.com/rust-lang/rust/issues/86698)
+- Moderate improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=5a8a44196b3cf099f8c9b0156bd902eaec0b4e5f&end=18073052d8c3544ccb73effd289ed3acda0d66c0&stat=instructions:u) (up to -1.4% on `incr-unchanged` builds of `ctfe-stress-4-check`)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=5a8a44196b3cf099f8c9b0156bd902eaec0b4e5f&end=18073052d8c3544ccb73effd289ed3acda0d66c0&stat=instructions:u) (up to 1.1% on `incr-unchanged` builds of `deeply-nested-async-opt`)
+- A perf run was done on the original PR that predicted a much better outcome here.
+- Left a comment asking PR author if they have any idea about the discrepancy.
```
Comment (rylev) [inline triage/2021-07-20.md:59]: nit: link https://github.com/rust-lang/rust/pull/86698#issuecomment-883647259

File: triage/2021-07-20.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+- [#86993 Replace associated item bound vars with placeholders when projecting](https://github.com/rust-lang/rust/pull/86993)
+- [#86777 Include terminators in instance size estimate](https://github.com/rust-lang/rust/pull/86777)
+- [#86588 Rollup of 8 pull requests](https://github.com/rust-lang/rust/pull/86588)
+- [#86034 Change entry point to 🛡️ against 💥 💥-payloads](https://github.com/rust-lang/rust/pull/86034)
+- [#84560 Inline Iterator as IntoIterator.](https://github.com/rust-lang/rust/pull/84560)
+
+#### Nags requiring follow up
+
+* [PR #87244 followup](https://github.com/rust-lang/rust/pull/87244#issuecomment-883635813)
+* [PR #86698 followup](https://github.com/rust-lang/rust/pull/86698#issuecomment-883647259)
```
Comment (rylev) [inline triage/2021-07-20.md:73]: I added `perf-regression` labels to these meaning that if they aren't addressed they'll show up in next week's untriaged pull request's section - which means they don't need to be here.

**Review by rylev — APPROVED** [review]
Looks great! I just made some comments, but there's no need to spend more time on this.

File: triage/2021-07-20.md (hunk 3 above)
Comment (pnkfelix) [reply triage/2021-07-20.md:14]: left [another comment](https://github.com/rust-lang/rust/pull/86993#issuecomment-884617720) that I hope captures what you were looking for, @rylev !

File: triage/2021-07-20.md (hunk 6 above)
Comment (pnkfelix) [reply triage/2021-07-20.md]: okay, removed the nags

File: triage/2021-07-20.md (hunk 5 above)
Comment (pnkfelix) [reply triage/2021-07-20.md]: thanks, added link!

File: triage/2021-07-20.md (hunk 4 above)
Comment (pnkfelix) [reply triage/2021-07-20.md]: thanks! added link.

**Pull request merged**